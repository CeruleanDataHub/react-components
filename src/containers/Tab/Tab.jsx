import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import { Typography } from "../Typography/Typography";
import { Icon } from "../Icon/Icon";

const Container = styled.div`
  display: flex;
  padding: 0.5em 1.5em;
  background: ${({ active }) => (active ? "white" : "#ddd")};
  border: 1px solid black;
  border-bottom-color: transparent;
  cursor: pointer;
  margin-right: 1em;
  border-radius: 3px 3px 0 0;
  align-items: center;
  justify-content: space-evenly;
`;

const Text = styled.span`
  font-size: 1em;
  color: black;
  user-select: none;
`;

export const Tab = ({ text, icon, active, onClick }) => (
  <Typography fontFamily="openSans">
    <Container active={active} onClick={onClick} data-active-tab-test={active}>
      {icon && <Icon name={icon} data-icon-test />}

      <Text>{text}</Text>
    </Container>
  </Typography>
);

Tab.propTypes = {
  text: PropTypes.string,
  icon: PropTypes.string,
  active: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

Tab.defaultProps = {
  text: "",
  icon: "",
  active: false
};
