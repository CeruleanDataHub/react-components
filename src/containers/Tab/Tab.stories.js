import { action } from "@storybook/addon-actions";
import React, { useState } from "react";
import styled from "styled-components";

import { Tab } from "./Tab";
import { Icon } from "../Icon/Icon";

export default {
  title: "Tab",
  component: Tab
};

const tabs = ["Tab 1", "Tab 2", "Tab 3"];

const TabsContainer = styled.div`
  display: flex;
  border-bottom: 1px solid black;
`;

const Tabs = () => {
  const [activeTab, setActiveTab] = useState("Tab 1");

  return (
    <TabsContainer>
      {tabs.map(tab => (
        <Tab
          text={tab}
          onClick={() => setActiveTab(tab)}
          active={activeTab === tab}
          key={tab}
        />
      ))}
    </TabsContainer>
  );
};

export const TabStory = () => (
  <div style={{ width: 100 }}>
    <Tab text="Tab" onClick={action("tab-clicked")} />
  </div>
);

export const TabActiveStory = () => (
  <div style={{ width: 100 }}>
    <Tab text="Tab" onClick={action("tab-clicked")} active />
  </div>
);

export const MultipleTabsStory = () => (
  <div>
    <Tabs />
  </div>
);

export const TabWithIconStory = () => (
  <div style={{ width: 200 }}>
    <Tab
      text="Tab with icon"
      onClick={action("tab-clicked")}
      icon={<Icon name="chef-hat" />}
    />
  </div>
);
