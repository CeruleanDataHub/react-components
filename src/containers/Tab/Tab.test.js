import { mount } from "enzyme";
import React from "react";

import { Tab } from "./Tab";

describe("Tab", () => {
  let component;
  let onClickMock;

  beforeEach(() => {
    onClickMock = jest.fn();
    component = mount(<Tab onClick={onClickMock} text="Some text" />);
  });

  it("renders", () => {
    expect(component.render()).toMatchSnapshot();
  });

  it("should not have icon", () => {
    expect(component.find("Icon[data-icon-test]")).not.toExist();
  });

  it("should not call onClick yet", () => {
    expect(onClickMock).not.toHaveBeenCalled();
  });

  it("when onClick is called, should call callback function", () => {
    component.props().onClick();

    expect(onClickMock).toHaveBeenCalled();
  });

  it("given active prop, is active", () => {
    component = mount(<Tab onClick={() => {}} active />);

    expect(component.find("div[data-active-tab-test=true]")).toExist();
  });

  it("given icon prop, has icon", () => {
    component = mount(<Tab onClick={() => {}} icon="chef-hat" />);

    expect(component.find("Icon[data-icon-test]")).toExist();
  });
});
