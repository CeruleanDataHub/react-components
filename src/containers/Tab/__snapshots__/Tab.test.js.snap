// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders 1`] = `
<div
  class="sc-AxjAm bYePpP"
  font-family="openSans"
>
  <div
    class="sc-AxhCb DQqhh"
    data-active-tab-test="false"
  >
    <span
      class="sc-AxhUy iAhtg"
    >
      Some text
    </span>
  </div>
</div>
`;
